internal/engine/docker_compose_build_and_deployer.go

```diff
@@ -12,58 +14,92 @@ import (
 
 type DockerComposeBuildAndDeployer struct {
 	dcc dockercompose.DockerComposeClient
+	ibd *ImageBuildAndDeployer
 }
 
 var _ BuildAndDeployer = &DockerComposeBuildAndDeployer{}
 
-func NewDockerComposeBuildAndDeployer(dcc dockercompose.DockerComposeClient) *DockerComposeBuildAndDeployer {
+func NewDockerComposeBuildAndDeployer(dcc dockercompose.DockerComposeClient,
+	ibd *ImageBuildAndDeployer) *DockerComposeBuildAndDeployer {
 	return &DockerComposeBuildAndDeployer{
 		dcc: dcc,
+		ibd: ibd,
 	}
 }
 
-// Extract the targets we can apply into DockerComposeTargets, or nil if we can't apply all targets.
-func (bd *DockerComposeBuildAndDeployer) extract(specs []model.TargetSpec) []model.DockerComposeTarget {
-	result := []model.DockerComposeTarget{}
+// Extract the targets we can apply -- DCBaD supports ImageTargets and DockerComposeTargets.
+func (bd *DockerComposeBuildAndDeployer) extract(specs []model.TargetSpec) ([]model.ImageTarget, []model.DockerComposeTarget) {
+	var iTargets []model.ImageTarget
+	var dcTargets []model.DockerComposeTarget
+
 	for _, s := range specs {
-		dc, isDC := s.(model.DockerComposeTarget)
-		if !isDC {
-			return nil
+		switch s := s.(type) {
+		case model.ImageTarget:
+			iTargets = append(iTargets, s)
+		case model.DockerComposeTarget:
+			dcTargets = append(dcTargets, s)
```

Comment:
should probably return nil if there's a spec it doesn't recognize, rather than quietly dropping it on the floor. see the other extract() method